service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -51,24 +56,28 @@ public Sample getSampleInStudy(String studyId, String sampleId) throws SampleNot
     }
 
     @Override
+    @PreAuthorize("hasPermission(#studyId, 'CancerStudy', 'read')")
     public BaseMeta getMetaSamplesOfPatientInStudy(String studyId, String patientId) {
 
         return sampleRepository.getMetaSamplesOfPatientInStudy(studyId, patientId);
     }
 
     @Override
+    @PreAuthorize("hasPermission(#studyIds, 'List<CancerStudyId>', 'read')")
     public List<Sample> fetchSamples(List<String> studyIds, List<String> sampleIds, String projection) {
 
         return sampleRepository.fetchSamples(studyIds, sampleIds, projection);
     }
 
     @Override
+    @PreAuthorize("hasPermission(#studyIds, 'List<CancerStudyId>', 'read')")
     public BaseMeta fetchMetaSamples(List<String> studyIds, List<String> sampleIds) {
 
         return sampleRepository.fetchMetaSamples(studyIds, sampleIds);
     }
 
     @Override
+    @PreAuthorize("hasPermission(#internalIds, 'List<InternalSampleId>', 'read')")
```

Comment:
we don't need to secure this one as it's used internally and don't correspond to an API (and it will never as we will never get internal ids from users), it's public because it's used by another service but it's internal. So you can delete this, and the part handles `List<InternalSampleId>` in evaluator, and also the new `getSampleByInternalId` from sample mapper and repository.